### PR TITLE
kimai, nixos/kimai: init at 2.24.0

### PR DESCRIPTION
**Pull request by peat-psuwit**
Kimai is a web-based multi-user time-tracking application. Works great for
everyone: freelancers, companies, organizations - everyone can track their
times, generate reports, create invoices and do so much more.

https://www.kimai.org/

The project is built using `buildComposerProject` helper. The module and the test are written following (primarily) Wordpress and Zoneminder.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ki/kimai/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  postInstall = ''
+    # Make available the console utility, as Kimai doesn't list this in
+    # composer.json.
+    mkdir -p "$out"/share/php/kimai "$out"/bin
+    ln -s "$out"/share/php/kimai/bin/console "$out"/bin/console
+  '';
+
+  passthru.tests = {
+    kimai = nixosTests.kimai;
+  }
```
Comment (OPNA2608) [inline pkgs/by-name/ki/kimai/package.nix:47]: Fails to parse here

```
error: syntax error, unexpected '=', expecting ';'
       at /home/runner/work/nixpkgs/nixpkgs/pkgs/by-name/ki/kimai/package.nix:49:8:
           48|
           49|   meta = {
             |        ^
           50|     description = "A web-based multi-user time-tracking application";
```

```suggestion
  };
```

File: pkgs/by-name/ki/kimai/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  meta = {
+    description = "A web-based multi-user time-tracking application";
+    homepage = "https://www.kimai.org/";
+    license = lib.licenses.agpl3Plus;
+    longDescription = "
+      Kimai is a web-based multi-user time-tracking application. Works great for
+      everyone: freelancers, companies, organizations - everyone can track their
+      times, generate reports, create invoices and do so much more.
+    ";
+    maintainers = lib.teams.php.members;
```
Comment (OPNA2608) [inline pkgs/by-name/ki/kimai/package.nix:58]: I don't *think* maintenance of this should just be assigned to the PHP team based on this software being written in PHP, but perhaps I'm unaware of outside information or PHP specific conventions here.

Usually you would first init yourself as a maintainer in [`/maintainers/maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/a5337b56fe644823e36acb7d59859c50056a377d/maintainers/maintainer-list.nix), then init this package with yourself as the maintainer.

File: pkgs/by-name/ki/kimai/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    mkdir -p "$out"/share/php/kimai "$out"/bin
+    ln -s "$out"/share/php/kimai/bin/console "$out"/bin/console
+  '';
+
+  passthru.tests = {
+    kimai = nixosTests.kimai;
+  }
+
+  meta = {
+    description = "A web-based multi-user time-tracking application";
```
Comment (OPNA2608) [inline pkgs/by-name/ki/kimai/package.nix:50]: https://github.com/NixOS/nixpkgs/blob/ce22e018581f4c6076d0612d0cb4b85c29872ee6/pkgs/README.md#meta-attributes

```suggestion
    description = "Web-based multi-user time-tracking application";
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Please format the completely new files with `nixfmt` from `nixfmt-rfc-style` for the CI formatting check to be happy.

File: pkgs/by-name/ki/kimai/package.nix (hunk 1 above)
Comment (peat-psuwit) [reply pkgs/by-name/ki/kimai/package.nix]: Well... the error was introduced when I was trying to add `passthru.tests` attribute last-minute. Should have tested the final version before submitting. Fixed.

File: pkgs/by-name/ki/kimai/package.nix (hunk 2 above)
Comment (peat-psuwit) [reply pkgs/by-name/ki/kimai/package.nix]: I was copy-pasting this area of snippet and didn't think about it much. Added my name to the maintainer list and set maintainers of the package and the test to myself.

File: pkgs/by-name/ki/kimai/package.nix (hunk 3 above)
Comment (peat-psuwit) [reply pkgs/by-name/ki/kimai/package.nix]: Fixed as suggested.

**Comment by OPNA2608** [comment]
Please rebase on master, x86_64-linux build on ofborg is failing due to missing https://github.com/NixOS/nixpkgs/commit/2f7b6eba4cf10bdebfbd3add70bbc2c0b36f64b6.

**Comment by peat-psuwit** [comment]
Rebased against current master. Please take another look.

**Comment by OPNA2608** [comment]
Can you resolve the merge conflict? I'll try to give this another review later this week.

**Comment by peat-psuwit** [comment]
Rebased and solved conflict. Please have another look.

**Review by OPNA2608 — APPROVED** [review]
LGTM.

**Review by OPNA2608 — COMMENTED** [review]
Ah, [branch-off for 24.11 already happened](https://github.com/NixOS/nixpkgs/issues/339153#issue-2501824991) so this won't make it into 24.11 via the master branch.

Are you fine with this going into stable 25.05 instead? Then please move the changelog entry to the [new 25.05 entries file](https://github.com/NixOS/nixpkgs/blob/86f8f6c6d2f136720bff5f25b16fedebd39f328f/nixos/doc/manual/release-notes/rl-2505.section.md).

Otherwise, we'll need to do a backport of this.

**Comment by peat-psuwit** [comment]
25.05 is OK for me. Changelog is moved to 25.05 entries.

**Review by OPNA2608 — APPROVED** [review]
LGTM!

**Pull request merged**